src/theory/logic_info.cpp

```diff
@@ -117,13 +117,25 @@ bool LogicInfo::isHigherOrder() const
   return d_higherOrder;
 }
 
-/** Is this the all-inclusive logic? */
-bool LogicInfo::hasEverything() const {
-  PrettyCheckArgument(d_locked, *this,
+bool LogicInfo::hasEverything() const
+{
+  PrettyCheckArgument(d_locked,
+                      *this,
                       "This LogicInfo isn't locked yet, and cannot be queried");
-  LogicInfo everything;
-  everything.lock();
-  return *this == everything;
+  // A logic has everything if all theories are enabled as well as quantifiers
```

Comment:
Can't we just do `everything.enableEverything(isHigherOrder())` and then compare equality to this?

Reply:
no because then a logic that does not have higher order enabled would fail the test. This would also not scale to the case of separation logic you mentioned.

Reply:
it would pass, and would be agnostic to sl.

Reply:
It wouldn't. To do as you suggest I'd have to create a new LogicInfo guy, set HOL to it, and return true when one of the comparisons to `everything` and `everythingAndHol` worked. It would indeed be agnostic to separation logic since that's a theory. 

The issue is that the equality operator requires that everything coincides. Since this method is testing just whether a logic *contains* everything, testing like I did seems more natural.